deploy/helm/sumologic/conf/logs/collector/otelcloudwatch/config.yaml

```diff
@@ -26,6 +26,33 @@ processors:
     send_batch_max_size: 2000
     send_batch_size: 1000
     timeout: 1s
+  transform/get_cloudwatch_pod:
+    log_statements:
+      - context: resource
+        statements:
+          - set(attributes["k8s.pod.name"], attributes["cloudwatch.log.stream"])
+          - set(attributes["k8s.container.name"], attributes["cloudwatch.log.stream"])
+          - set(attributes["k8s.namespace.name"], attributes["cloudwatch.log.stream"])
+          - replace_pattern(attributes["k8s.pod.name"], "^.*kube\\.var\\.log\\.containers\\.([0-9a-zA-Z\\-]+)\\_.*", "$$1")
+          - replace_pattern(attributes["k8s.container.name"], "^.*kube\\.var\\.log\\.containers\\.([0-9a-zA-Z\\-]+)\\_.*", "$$1")
+          - replace_pattern(attributes["k8s.namespace.name"], "^.*kube\\.var\\.log\\.containers\\.[0-9a-zA-Z\\-]+\\_([a-zA-Z\\-]*)_.*", "$$1")
+  transform/parse:
+    log_statements:
+      - context: log
+        statements:
+          - set(body, ParseJSON(body)) where IsMatch(body, "^{") == true
+  transform/flatten:
+    log_statements:
+      - context: log
+        statements:
+          - merge_maps(attributes, body, "insert")
+          - set(body, "") where IsMatch(body, "^{") == true
+  transform/move:
+    log_statements:
+      - context: log
+        statements:
+          - set(attributes["k8s.container.name"], resource.attributes["k8s.container.name"])
+          - set(attributes["k8s.deployment.name"], resource.attributes["k8s.deployment.name"])
```

Comment:
Couldn't we do this directly in `transform/get_cloudwatch_pod`? And why aren't we moving `k8s.pod.name` as well?

Reply:
It was split because of different contexts (resource vs log) but maybe this can be done instead
{code}
transform/cloudwatch_logs:
    error_mode: ignore
    log_statements:
      - context: resource
        statements:
          - set(attributes["k8s.pod.name"], attributes["cloudwatch.log.stream"])
          - set(attributes["k8s.container.name"], attributes["cloudwatch.log.stream"])
          - set(attributes["k8s.namespace.name"], attributes["cloudwatch.log.stream"])
          - replace_pattern(attributes["k8s.pod.name"], "^.*kube\\.var\\.log\\.containers\\.([0-9a-zA-Z\\-]+)\\_.*", "$$1")
          - replace_pattern(attributes["k8s.container.name"], "^.*kube\\.var\\.log\\.containers\\.([0-9a-zA-Z\\-]+)\\_.*", "$$1")
          - replace_pattern(attributes["k8s.namespace.name"], "^.*kube\\.var\\.log\\.containers\\.[0-9a-zA-Z\\-]+\\_([a-zA-Z\\-]*)_.*", "$$1")
      - context: log
        statements:
          - set(attributes["k8s.container.name"], resource.attributes["k8s.container.name"])
          - set(attributes["k8s.pod.name"], resource.attributes["k8s.pod.name"])
          - set(attributes["k8s.deployment.name"], resource.attributes["k8s.deployment.name"])
  transform/parse:
    log_statements:
      - context: log
        statements:
          - set(body, ParseJSON(body)) where IsMatch(body, "^{") == true
          - merge_maps(attributes, body, "insert")
          - set(body, "") where IsMatch(body, "^{") == true
 {code}

Reply:
I'm actually a bit confused. Why do we even need to set record attributes here? Why is the second group of statements even necessary? These keys are supposed to be resource attributes to begin with.

Reply:
We need to set the container name, because the container pipeline in the metadata layer, looks for the container name as a record attribute

Reply:
yes, I believe the metadata pipeline moves the following record attributes to the resource level

```
groupbyattrs/containers:
    keys:
      - k8s.container.id
      - k8s.container.name
      - k8s.namespace.name
      - k8s.pod.name
      - _collector
  ```

Reply:
this change is completed